In [1]:
import pandas as pd
import numpy as np

In [13]:
df = pd.read_csv("Data.csv")
df.columns = ["SalesPerson","Old_Scheme","New_Scheme"]
df["Old_Scheme_in_thousands"] = df["Old_Scheme"] * 1000
df["New_Scheme_in_thousands"] = df["New_Scheme"] * 1000
df

SalesPerson  Old_Scheme  New_Scheme  Old_Scheme_in_thousands  \
0             1          57          62                    57000   
1             2         103         122                   103000   
2             3          59          54                    59000   
3             4          75          82                    75000   
4             5          84          84                    84000   
5             6          73          86                    73000   
6             7          35          32                    35000   
7             8         110         104                   110000   
8             9          44          38                    44000   
9            10          82         107                    82000   
10           11          67          84                    67000   
11           12          64          85                    64000   
12           13          78          99                    78000   
13           14          53          39                    53000   
14           15          41          34                    41000   
15           16          39          58                    39000   
16           17          80          73                    80000   
17           18          87          53                    87000   
18           19          73          66                    73000   
19           20          65          78                    65000   
20           21          28          41                    28000   
21           22          62          71                    62000   
22           23          49          38                    49000   
23           24          84          95                    84000   
24           25          63          81                    63000   
25           26          77          58                    77000   
26           27          67          75                    67000   
27           28         101          94                   101000   
28           29          91         100                    91000   
29           30          50          68                    50000   

    New_Scheme_in_thousands  
0                     62000  
1                    122000  
2                     54000  
3                     82000  
4                     84000  
5                     86000  
6                     32000  
7                    104000  
8                     38000  
9                    107000  
10                    84000  
11                    85000  
12                    99000  
13                    39000  
14                    34000  
15                    58000  
16                    73000  
17                    53000  
18                    66000  
19                    78000  
20                    41000  
21                    71000  
22                    38000  
23                    95000  
24                    81000  
25                    58000  
26                    75000  
27                    94000  
28                   100000  
29                    68000

Find the mean of old scheme and new scheme column.

In [14]:
df.Old_Scheme_in_thousands.mean()

68033.33333333333

In [15]:
df.New_Scheme_in_thousands.mean()

72033.33333333333

In [16]:
diffmean = df["New_Scheme_in_thousands"].mean() - df["Old_Scheme_in_thousands"].mean()
diffmean

4000.0

Question 2 :Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (8 points)

In [17]:
from scipy import stats 
import numpy as np

In [18]:
s,p = stats.ttest_ind(df.Old_Scheme_in_thousands, df.New_Scheme_in_thousands)

In [22]:
p

0.49063515686248105

In [23]:
p<0.05

False

Question 3 : What conclusion does the test (p-value) lead to?

Our p-value is 0.49 which is greater than the common significance value of 0.05. Since it is greater, we accept Null Hypothesis.

Question 4A

When the null hypothesis is true and you reject it, you make a type I error. The probability of making a type I error is alpha, which is the level of significance you set for your hypothesis test. P(Type1 Error) = 0.05

In [ ]:
Question 4B

m2 = Average sums assured by salesperson in new scheme.

m1 = Average sums assured by salesperson in old scheme

H0: m2-m1 ≤ 5000 (statistially) H1: m2-m1 > 5000

In [24]:
from numpy import mean
from math import sqrt
from scipy.stats import t
def dependent_ttest(data1, data2, alpha,meanDif):
	n = len(data1)
	d1 = sum([(data1[i]-data2[i])**2 for i in range(n)])
	d2 = sum([data1[i]-data2[i] for i in range(n)])
	sd = sqrt((d1 - (d2**2 / n)) / (n - 1))
	sed = sd / sqrt(n)
	t_stat = (meanDif) / sed
	df = (n - 1)
	cv = t.ppf(1.0 - alpha, df)
	p = (1.0 - t.cdf(abs(t_stat), df)) * 2.0
	return t_stat, cv, p,sed
t_stat, cv, p,sed=dependent_ttest(df['New_Scheme_in_thousands'],df['Old_Scheme_in_thousands'],0.05,5000)
t_stat, cv, p,sed

(1.9448929779430468,
 1.6991270265334972,
 0.06154216863016693,
 2570.8355455569017)

Question 4C

m2 = Average sums assured by salesperson in new scheme.

m1 = Average sums assured by salesperson in old scheme H0: m2-m1 = 4000 (statistially) H1: m2-m1 >= 0

H0 will be rejected if test statistics > t_critical.

alpha = 0.05 degree of Freedom = 29, critical value for t statistic will be 1.699.

In [25]:
xbar= cv*sed
xbar

4368.176156228719

Hence, H0 will be rejected when xbar ≥ 4368.17

Probability(Type 2 error) =P (xbar < 4368.176)

In [26]:
t1 = (xbar-5000)/sed
p =1- stats.t.cdf(t1,29)
p

0.5962027475205683

In [27]:
p =1- stats.t.cdf(t1,29)
p

0.5962027475205683

Power of test = 1-p

In [29]:
1-p

0.40379725247943166